libgssapi/src/util.rs

```diff
@@ -236,13 +237,25 @@ impl Deref for Buf {
     type Target = [u8];
 
     fn deref(&self) -> &Self::Target {
-        unsafe { slice::from_raw_parts(self.0.value.cast(), self.0.length as usize) }
+        unsafe {
+            if self.0.value.is_null() {
```

Comment:
This should check that both the length is 0 and the value is null. If the length isn't 0 and the value is null then we would like slice::from_raw_parts to fail

Reply:
Good call, added